python/lsst/daf/butler/script/queryDataIds.py

```diff
@@ -89,8 +113,8 @@ def queryDataIds(repo, dimensions, datasets, where, collections, order_by, limit
             offset = None
         results.limit(limit, offset)
 
-    if len(results.graph) > 0:
+    if results.count() > 0 and len(results.graph) > 0:
         table = _Table(results)
-        return table.getAstropyTable(not order_by)
+        return table.getAstropyTable(not order_by), None
     else:
-        return None
+        return None, "\n".join(results.explain_no_results())
```

Comment:
Is this tuple the convention for returning no results explanations elsewhere in the butler? I feel like raising `NoResults` (or some other custom exception) with the explanation in the payload would be more pythonic. Looking at the tests, it certainly would simplify this interface.